assignments/assignment06.rb

```diff
@@ -0,0 +1,282 @@
+# ========================================================================================
+# Assignment 6
+# ========================================================================================
+
+# ========================================================================================
+#  Problem 1 - PriorityQueue
+
+# implement a PriorityQueue, validate using MiniTest unit tests
+
+class PriorityQueue
+
+  class Node
+    attr_accessor :item, :link
+    def initialize(item, link)
+      @item = item
+      @link = link
+    end
+  end
+
+  def initialize
+    # your implementation here
+    @hi_q = nil
+    @md_q = nil
+    @lo_q = nil
+  end
+
+  def enqueue(item, priority=:medium)
+    # your implementation here
+    case priority
+    when :high
+      @hi_q = Node.new(item, @hi_q)
+    when :medium
+      @md_q = Node.new(item, @md_q)
+    when :low
+      @lo_q = Node.new(item, @lo_q)
+    else
+      puts "Error: unknown priority, #{item} not enqueued"
+    end
+  end
+
+  def dequeue
```

Comment:
You have a LOT of repeated code here. Can you think of a way of eliminating this repetition?